### PR TITLE
Speed up CI

### PR DESCRIPTION
**Pull request by mhsdef**
Most of our jobs are quite fast, but, there's a few that annoy...

### CODE REVIEW

**Comment by mhsdef** [comment]
It takes a minute to install our node dependencies. 💀 

Screenshot 2024-09-25 at 2 34 12 PM

Screenshot 2024-09-25 at 2 34 20 PM


The VIP action (first screenshot) tries cache hits but seems to be missing every time right now.

**Review by chriszarate — DISMISSED** [review]
Nice sleuthing! If the [VIP action](https://github.com/Automattic/vip-actions/tree/trunk/nodejs-setup) is missing cache unnecessarily, can you fix it? It's used in quite a few places so it would be very beneficial

**Comment by mhsdef** [comment]
I was still investigating... 😅 

Yes, that's on the table, if it makes sense.

**Comment by mhsdef** [comment]
It's not misses. It's some sort of weird timeout-ish behavior with `actions/cache@v4` + how we're using it in the vip action. The runner just hangs around action entry and/or exit from what I've seen so far.

```
2024-09-25T19:13:52.3783534Z > remote-data-blocks@0.1.0 check-types
2024-09-25T19:13:52.3784116Z > tsc --noEmit
2024-09-25T19:13:52.3784324Z 
2024-09-25T19:14:05.1314013Z Post job cleanup.
2024-09-25T19:14:58.3901707Z Post job cleanup.                            <--------
2024-09-25T19:14:58.5816394Z Post job cleanup.
2024-09-25T19:14:58.6958727Z Cache hit occurred on the primary key Linux-node-v20.17.0-d691fdb1a5b3dafdb890b8f3ac075f180c215a05e054d46c032bef68a654a75b, not saving cache.
2024-09-25T19:14:58.7041176Z Post job cleanup.
2024-09-25T19:14:58.8430234Z Cache hit occurred on the primary key node-cache-Linux-npm-ea13e6b46516f91ea8b0896a527ec875b8765def3bce0d11ede2ab9b14224f2f, not saving cache.
2024-09-25T19:14:58.8526349Z Post job cleanup.
```

```
blocks --use-compress-program unzstd
2024-09-25T19:12:25.0930469Z Received 186192165 of 186192165 (100.0%), 177.6 MBs/sec
2024-09-25T19:12:34.4683585Z Cache restored successfully
2024-09-25T19:12:34.5355263Z Cache restored from key: Linux-node-v20.17.0-d691fdb1a5b3dafdb890b8f3ac075f180c215a05e054d46c032bef68a654a75b
2024-09-25T19:13:32.0156193Z ##[group]Run actions/cache@v4.                <--------
2024-09-25T19:13:32.0156472Z with:
2024-09-25T19:13:32.0156675Z   path: .cache/jest
2024-09-25T19:13:32.0157153Z   key: Linux-node-jest-2cd0ad1b58bbd793d18ff805791bb97b02d0a2f769fc088457f49e6676b6a6f1
```

File: package.json [hunk 1]
```diff
@@ -79,7 +78,6 @@
     "eslint": "8.57.1",
     "fork-ts-checker-webpack-plugin": "9.0.2",
     "happy-dom": "15.7.4",
-    "phplint": "2.0.5",
```
Comment (mhsdef) [inline package.json:82]: Psalm covers this now.

**Pull request merged**